### ISSUE TITLE
Support multiple environments

### ISSUE THREAD

**Issue by TheAnachronism** [twpayne/chezmoi#2021]
## Is your feature request related to a problem? Please describe.

It kind of is.
My ideal use-case for dotfile managers would be for multiple environments. As an example:
I have my own shell configurations for bash and some configuration for client tools like `kubectl`. The bash configuration I'd like to use on all the machines I work on, but the configuration for kubectl I only need on personal / home machines.

## Describe the solution you'd like

Maybe a repository structure like this:
- Root directory with configurations / dotfiles that get applied for all environments
- different directories for the different environments a user can create
  - To clearly seperate these environment directories from other directories maybe a prefix of `env_` can be used
  - Or have the different environments inside a `environments` directory

## Describe alternatives you've considered

I've had a look at how problems are described / solved for issues like:
https://github.com/twpayne/chezmoi/issues/780
https://github.com/twpayne/chezmoi/issues/1234

## Additional context

In summary, a bit:
A way to have one repository but be able to support multiple working-environments / profiles.

**Comment by TheAnachronism** [comment]
Another example I missed earlier:

On my server environments, I'd like to quickly add the SSH public-keys to my `authorized_hosts` file.
Maybe I do this the wrong in the case, but until now, I had my own setup script that downloaded these public-keys from a server and add them to the correct file.

This is only needed for server environments, but now home/personal machines.

**Comment by twpayne** [comment]
chezmoi's philosophy is that the dotfiles for all machines are generated from a single source of truth. See [this comment](https://github.com/twpayne/chezmoi/issues/1992#issuecomment-1087890122) for more detail on how to handle differences from machine to machine.
